### PR TITLE
fix(copy): treat null inferred columns as Utf8 to preserve later values

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This PR:

- Normalize inferred file schemas so columns inferred as `Null` are treated as `Utf8`, preventing later non-null values from being dropped when `schema_infer_max_record` is too small.
- Ensure JSON export preserves explicit nulls so null-only columns remain visible during schema inference.
- Add CSV/JSON copy-from-fs tests where a column is null for the first N rows and non-null in the last row, validating import with `schema_infer_max_record = N-1`.

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [ ] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.
- [ ] API changes are backward compatible.
- [ ] Schema or data changes are backward compatible.

### CODE REVIEW

**Comment by WenyXu** [comment]
@codex review

File: src/common/datasource/src/util.rs [hunk 1]
```diff
@@ -27,9 +29,30 @@ pub fn find_dir_and_filename(path: &str) -> (String, Option<String>) {
...
+                f.clone()
+            }
+        })
+        .collect::<Vec<_>>();
+
+    arrow_schema::Schema {
+        fields: arrow_schema::Fields::from(fields),
+        metadata: schema.metadata,
+    }
+}
```
Comment (evenyag) [inline src/common/datasource/src/util.rs:52]: Should we add an option to return error if can't infer a non null schema? Maybe we should allow users specifying the column schema in the future.

Comment (WenyXu) [reply src/common/datasource/src/util.rs:52]: The upper layer will return an error if it attempts to insert a null value into a non-nullable column. Here, we only focus on normalizing the inferred schema.
